bedrock-codec/src/main/java/org/cloudburstmc/protocol/bedrock/codec/BedrockCodec.java

```diff
@@ -129,16 +128,13 @@ public <T extends BedrockPacket> Builder registerPacket(Class<T> packetClass, Be
             checkArgument(id >= 0, "id cannot be negative");
             checkArgument(!packets.containsKey(packetClass), "Packet class already registered");
 
-            Supplier<Object> factory;
-            try {
-                MethodHandles.Lookup lookup = MethodHandlesExtensions.privateLookupIn(packetClass, MethodHandles.lookup());
-                MethodHandle handle = lookup.findConstructor(packetClass, MethodType.methodType(void.class));
-                factory = LambdaFactory.createSupplier(handle);
-            } catch (NoSuchMethodException | IllegalAccessException e) {
-                throw new IllegalArgumentException("Unable to find suitable constructor for packet factory", e);
-            }
-
-            BedrockPacketDefinition<T> info = new BedrockPacketDefinition<>(id, (Supplier) factory, serializer);
+            BedrockPacketDefinition<T> info = new BedrockPacketDefinition<>(id, () -> {
+                try {
+                    return packetClass.newInstance();
```

Comment:
We used the previous library instead of Reflection because of its slowdown. Maybe we could use ASM to construct a Supplier from bytecode.

Reply:
What about doing something like `registerPacket(LoginPacket::new, LoginSerializer_v291.INSTANCE, 1)` instead?

Reply:
But i have heard that java reflection will also generate class to make the process faster https://stackoverflow.com/a/22905873/16676567
> Have a look at [Reflection sources](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/43cb25339b55/src/share/classes/sun/reflect) in OpenJDK, MethodAccessorGenerator.java in particular. In short, java.lang.Method has an accessor to which invocations are delegated. First 15 calls of Method.invoke() are executed through a native call by JVM (see NativeMethodAccessorImpl.java). Then, as soon as sun.reflect.inflationThreshold is reached (15 by default), bytecode generation runs and NativeMethodAccessorImpl is replaced with dynamically generated Java accessor.